hs_app_timeseries/models.py

```diff
@@ -23,16 +63,48 @@ class Site(AbstractMetaDataElement):
     def __unicode__(self):
         return self.site_name
 
-    class Meta:
-        # site element is not repeatable
-        unique_together = ("content_type", "object_id")
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        element = cls.objects.get(id=element_id)
+
+        # if the user has entered a new elevation datum
+        elevation_datum_str = 'elevation_datum'
+        if elevation_datum_str in kwargs:
+            if element.elevation_datum != kwargs[elevation_datum_str]:
+                # check if the user has entered a new name for elevation datum
+                if kwargs[elevation_datum_str] not in [item.name for item in
+                                                       element.metadata.cv_elevation_datums.all()]:
+                    # generate term for the new name
+                    kwargs[elevation_datum_str] = kwargs[elevation_datum_str].strip()
+                    term = _generate_term_from_name(kwargs[elevation_datum_str])
+                    elevation_datum = CVElevationDatum.objects.create(
+                        metadata=element.metadata, term=term, name=kwargs[elevation_datum_str])
+                    elevation_datum.is_dirty = True
+                    elevation_datum.save()
```

Comment:
This block of code seems to be effectively duplicated quite a few times.  Perhaps a utility function could cut down on the boilerplate?